cmd/main.go

```diff
@@ -744,158 +452,448 @@ func Run() error {
+	threadAddSchema     := threadAddCmd.Flag("schema", "Thread schema ID. Supersedes schema filename").String()
+	threadAddSchemaFile := threadAddCmd.Flag("schema-file", "Thread schema filename, supersedes the built-in schema flags").String() // @note could be swapped to .File() perhaps
+	threadAddBlob       := threadAddCmd.Flag("blob", "Use the built-in blob schema for generic data").Bool()
+	threadAddCameraRoll := threadAddCmd.Flag("camera-roll", "Use the built-in camera roll schema").Bool()
+	threadAddMedia      := threadAddCmd.Flag("media", "Use the built-in media schema").Bool()
+	cmds[threadAddCmd.FullCommand()] = func () error {
 		return ThreadAdd(*threadAddName, *threadAddKey, *threadAddType, *threadAddSharing, *threadAddWhitelist, *threadAddSchema, *threadAddSchemaFile, *threadAddBlob, *threadAddCameraRoll, *threadAddMedia)
+	}
 
-	case threadListCmd.FullCommand():
+	// thread list
+	threadListCmd := threadCmd.Command("list", "Lists info on all threads").Alias("ls").Default()
+	cmds[threadListCmd.FullCommand()] = func () error {
 		return ThreadList()
+	}
 
-	case threadGetCmd.FullCommand():
+	// thread get
+	threadGetCmd      := threadCmd.Command("get", "Gets and displays info about a thread")
+	threadGetThreadID := threadGetCmd.Arg("thread", "Thread ID").Required().String()
+	cmds[threadGetCmd.FullCommand()] = func () error {
 		return ThreadGet(*threadGetThreadID)
+	}
 
-	case threadDefaultCmd.FullCommand():
-		return ThreadDefault()
-
-	case threadPeerCmd.FullCommand():
+	// thread peer
+	threadPeerCmd      := threadCmd.Command("peer", "Lists all peers in a thread").Alias("peers")
+	threadPeerThreadID := threadPeerCmd.Arg("thread", "Thread ID").Required().String()
+	cmds[threadPeerCmd.FullCommand()] = func () error {
 		return ThreadPeer(*threadPeerThreadID)
+	}
 
-	case threadRenameCmd.FullCommand():
+	// thread rename
+	threadRenameCmd      := threadCmd.Command("rename", "Renames a thread. Only the initiator of a thread can rename it.").Alias("mv")
+	threadRenameThreadID := threadRenameCmd.Arg("thread", "Thread ID").Required().String()
+	threadRenameName     := threadRenameCmd.Arg("name", "The name to rename the thread to").Required().String()
+	cmds[threadRenameCmd.FullCommand()] = func () error {
 		return ThreadRename(*threadRenameName, *threadRenameThreadID)
+	}
 
-	case threadUnsubscribeCmd.FullCommand():
-		return ThreadUnsubscribe(*threadUnsubscribeThreadID)
+	// thread abandon
+	threadAbandonCmd      := threadCmd.Command("abandon", "Abandon a thread. If no one is else remains participating, the thread dissipates.").Alias("unsubscribe").Alias("leave").Alias("remove").Alias("rm")
+	threadAbandonThreadID := threadAbandonCmd.Arg("thread", "Thread ID").Required().String()
+	cmds[threadAbandonCmd.FullCommand()] = func () error {
+		return ThreadAbandon(*threadAbandonThreadID)
+	}
+
+	// thread snapshot
+	// @todo are snapshots blocks?
```

Comment:
Nope, they're an encrypted object containing thread metadata and the latest block hash, which is enough to recover the thread.